### PR TITLE
Publish Event resource creation/deletion to a Kafka topic

### PR DESCRIPTION
**Pull request by SEQUOIIA**
What has changed:
- Publishing messages to a Kafka topic is now possible
- By optionally setting "enablePublisher" to true, a background service will spawn that waits for messages to be sent to it. Upon receiving a message/payload, said service will wrap that message in a message "enclosure" with extra metadata.
- By optionally setting "enablePublisher" to true, Inventa look for creation and deletion of Event objects, and send metadata related to these as a message/payload to the previously mentioned background service

Out of the box, all these additions are disabled by default.

No need to merge this PR right away. Just looking for someone else's input.

### CODE REVIEW

File: operator/misc/publisher.go [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+	}
+
+	for {
+		msg := <-p.messageChannel
+
+		go func() {
+			enclosure := MessageEnclosure{
+				App:            "inventa-operator",
+				Timestamp:      strconv.FormatInt(time.Now().Unix(), 10),
+				ClusterName:    "hellman",
```
Comment (rifisdfds) [inline operator/misc/publisher.go:153]: I don't think we should hard-code the ClusterName

File: operator/misc/publisher.go [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	for {
+		msg := <-p.messageChannel
+
+		go func() {
+			enclosure := MessageEnclosure{
+				App:            "inventa-operator",
```
Comment (rifisdfds) [inline operator/misc/publisher.go:151]: Could we perhaps retrieve/determine this name from the operator/deployment rather than hard-code?

**Review by rifisdfds — CHANGES_REQUESTED** [review]
Looks good with a glance over. I've added a couple of suggestions

File: operator/misc/publisher.go (hunk 1 above)
Comment (SEQUOIIA) [reply operator/misc/publisher.go:153]: Aye, definitely a placeholder that I forgot about. 👍

File: operator/misc/publisher.go (hunk 2 above)
Comment (SEQUOIIA) [reply operator/misc/publisher.go:151]: Could do one of these:
- https://stackoverflow.com/a/37261036
- https://stackoverflow.com/a/37253342

Any particular preference? (I'm fine with either myself)

Comment (rifisdfds) [reply operator/misc/publisher.go:151]: `/etc/hostname` gets the job done quickest I assume, even though there is the potential for it to be `localhost` in some environments. The [pod info](https://kubernetes.io/docs/tasks/inject-data-application/downward-api-volume-expose-pod-information/) method might be technically better but more complex

I'm happy with either as well, we can always change it later

**Pull request closed without merging**